flips/20220203-capability-controllers.md

```diff
@@ -0,0 +1,341 @@
+# Capability controllers
+
+| Status        | Proposed                                           |
+| :------------ | :------------------------------------------------- |
+| **FLIP #**    | [798](https://github.com/onflow/flow/pull/798)     |
+| **Author(s)** | Janez Podhostnik (janez.podhostnik@dapperlabs.com) |
+| **Updated**   | 2022-02-03                                         |
+
+## Preface
+
+Cadence encourages a [capability-based security](https://en.wikipedia.org/wiki/Capability-based_security) model as described on the Flow [doc site](https://docs.onflow.org/cadence/language/capability-based-access-control). Capabilities are themselves a new concept that most Cadence programmers need to understand, but the API for syntax around Capabilities (especially the notion of “links” and “linking”), and the associated concepts of the public and private storage domains, lead to Capabilities being even more confusing and awkward to use. This proposal suggests that we could get rid of links entirely, and replace them with Capability Controllers (henceforth referred to as CapCons) which could make Capabilities easier to understand, and easier to work with.
+
+The following is a quick refresher of the current state of the Capabilities API (from the [flow doc site](https://docs.onflow.org/cadence/language/capability-based-access-control)).
+
+### Example Resource definition
+
+In the following examples let's assume that the following interface and resource type that implements the interface are defined.
+
+```cadence
+pub resource interface HasCount {
+   pub count: Int
+}
+
+pub resource Counter: HasCount {
+   pub var count: Int
+
+   pub init(count: Int) {
+       self.count = count
+   }
+}
+```
+
+The _issuer_ (`AuthAccount`) has also created an instance of this resource and saved it in its private storage.
+
+```cadence
+AuthAccount.save(<-create Counter(count: 42), to: /storage/counter)
+```
```

Comment:
This code is not valid, `AuthAccount` is a built-in type with a built-in constructor function of the same name.
Code examples below also refer to `authAccount`, which is maybe the same? 

Instead of naming the variable after what type it has, name it after the role / function, e.g. `issuer`

```suggestion
The _issuer_, `issuer` of type `AuthAccount`, has also created an instance of this resource and saved it in its private storage.

```cadence
AuthAccount.save(<-create Counter(count: 42), to: /storage/counter)
```
```

Reply:
`issuer.save(<-create Counter(count: 42), to: /storage/counter)` it should be I guess